src/main/java/org/mule/service/http/impl/service/HttpServiceImplementation.java

```diff
@@ -70,8 +77,17 @@ public HttpServerFactory getServerFactory() {
   }
 
   @Inject
-  public HttpServerFactory getServerFactory(@Named(OBJECT_MULE_CONTEXT) MuleContext muleContext) {
-    return new ContextHttpServerFactoryAdapter(muleContext.getId(), listenerConnectionManager);
+  public HttpServerFactory getServerFactory(Registry registry) {
+    Optional<String> appName = registry.lookupByName(APP_NAME_PROPERTY);
+    Optional<String> domainName = registry.lookupByName(DOMAIN_NAME_PROPERTY);
+    if (domainName.isPresent() && appName.isPresent()) {
+      return new ContextHttpServerFactoryAdapter(appName.get(), domainName.get(), listenerConnectionManager);
+    }
+    return domainName
+        .map(d -> new ContextHttpServerFactoryAdapter(d, listenerConnectionManager))
+        .orElseGet(() -> appName.map(a -> new ContextHttpServerFactoryAdapter(a, listenerConnectionManager))
+            .orElseThrow(() -> new MuleRuntimeException(createStaticMessage("Could not create server factory because neither "
+                + APP_NAME_PROPERTY + " nor " + DOMAIN_NAME_PROPERTY + " is defined"))));
```

Comment:
`are defined`

Reply:
I also had doubts with this, but i think this is the correct form.
https://english.stackexchange.com/questions/1144/neither-michael-nor-albert-is-correct-or-neither-michael-nor-albert-are-corre